crates/subspace-farmer/src/farming.rs

```diff
@@ -107,6 +107,14 @@ struct Salts {
     next: Option<Salt>,
 }
 
+struct AbortOnDrop<T>(JoinHandle<T>);
```

Comment:
There is a good wrapper in the `rust-analyzer` which is called `Defer` in go like manner. It has only constructor - a `FnOnce` closure which will be called on drop. Should we try to use smth like this?

Reply:
Not sure what you mean by that

Reply:
I mean that it would remove those class of struct overall. You can just call `let _handle = Defer::new(|| handle.abort());`.
It would remove unnecessary boilerplate with implementing drop every time.

Reply:
If we had that use case more often I think it'd be fine, but we have just one-off here. If you notice another use case for this, feel free to create a more general purpose utility.